### ISSUE TITLE
Mobile agent help with Provisioning

### ISSUE THREAD

**Issue by agentcooper75** [hyperledger/aries-vcx#322]
Hej, I am building a mobile iOS agent and already spend some time on the 3rd party iOS Swift wrapper.
After updating the vcx library, I noticed that many parts of the provisioning have been deprecated/removed from the library, so the demo isn‘t working anymore.

So I started my own Swift wrapper, but uncertain about the correct workflow. Based on #209 I tried to figure out the correct workflow. So far I have working:

- createWallet
- openMainWallet
- vcxProvisionCloudAgent (connection to dummy cloud agent)
- closeMainWallet

Now that "vcx_init_core" is not working anymore, how to init core with the returned agency config? 
Digging deeper into #210 it's supposed to continue with:
- vcx_init_ffi **(not available in iOS wrapper. Is this vcxInitThreadpool?)**
- vcx_init_logger
- vcx_init_issuer_configuration **(not required for mobile agent)**
- vcx_configure_issuer_wallet **(not required for mobile agent)**
- vcx_create_agency_client(agency_config) - **vcxCreateAgencyClient**
- openMainWallet
- vcx_open_pool_directly(pool_config) - **vcxOpenMainPool (connection to indy pool run in a Docker container)**
This currently returns an error ("Connection to Pool Ledger.").

Thanks

**Comment by Patrik-Stas** [comment]
Hi @agentcooper75!

sorry for coming to you bit late, also sorry for the confusion about these init functions - it's been a bit battleground since we forked the project and init API was one of the things we reworked significantly and as you've discovered, seems like we've left some legacy pieces left behind in the wrappers.

I really appreciate you creating this issue as this can help us refine the onboarding and making this easier for new-comers. We didn't many new external consumers, but I do feel recently `aries-vcx` is getting some traction.

You've got the things almost right, but let me distill this for you a bit more. On mobile (or through any FFI), the first things you should always start with are:
`Stage A`:
1. `vcx_init_threadpool`
2. `vcx_init_logger`

Continuing next, it depends whether you are running your application the first time or not - more precisely, whether you have previously. So if you are running the first time, you need to execute
`Stage B`:
1. create wallet `vcx_create_wallet`
2. open wallet `vcx_open_main_wallet`
3. provisioned cloud agent (eg. "create account in mediator agency") `vcx_provision_cloud_agent` (this returns JSON we call `agency client config`. You need to store this piece of information somewhere persistently, because you will need it in the following steps, as you'll see below)
4. close wallet `vcx_close_main_wallet`

Continuing next, the flow is the same whether you run the first time or not. You need configure library memory state running 
`Stage C`:
1. open wallet `vcx_open_main_wallet`
2. create agency client `vcx_create_agency_client_for_main_wallet` (passing in `agency client config` as parameter)
3. open connection to indy pool `vcx_open_main_pool`

### To sum it up:
- The first time you run you execute all stages `A, B, C`. The order is important (also within each stage).
- Next time you startup, you only run `A, C`.

**Comment by agentcooper75** [comment]
@Patrik-Stas Thank you for the detailed information. 
Your description helped me to provision the mobile agent on iOS using the latest release 0.20.0

**Comment by Patrik-Stas** [comment]
That's great! I'll close this issue then, if you run into an issue, feel free to also get in touch with us at #aries-vcx channgel on Hyperledger RocketChat.
